### PR TITLE
docs: team sync

### PR DESCRIPTION
**Pull request by xlson**
**What this PR does / why we need it**:

Docs for team sync

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/auth/auth-proxy.md [hunk 1]
```diff
@@ -218,3 +218,18 @@ ProxyPassReverse / http://grafana:3000/
...
+>  Only available in Grafana Enterprise v6.3+
+
+With Team Sync you can feed team ID's to Grafana and map them to Teams in Grafana. This allows you to automatically put your users into specific teams. 
+
+To do this, you need to configure the optional `Groups` header and use that to send your team information.
+
+
+```bash
+# Example `headers = Groups:X-WEBAUTH-GROUPS`
+headers =
```
Comment (torkelo) [inline docs/sources/auth/auth-proxy.md]: this example is a bit strange, with the example in a comment and the uncomented line empty. Do we need to mention here how to get the team ids? is that Grafana internal team ids?

Comment (xlson) [reply docs/sources/auth/auth-proxy.md]: Yeah, you're right. We should add more detail, showing real config and an example request using `curl` or such that uses the feature.

Comment (apple-fan-pl) [reply docs/sources/auth/auth-proxy.md]:
```suggestion
headers = fuck-kde
```

Comment (gotjosh) [reply docs/sources/auth/auth-proxy.md]: @torkelo could I please get a second look at this? I have added examples and reworded a the bit of auth proxy section.

**Comment by gotjosh** [comment]
I'm not sure if this intended behaviour and I deliberately avoided going down a rabbit hole. But seems like:

https://github.com/grafana/grafana/blob/f3f03ceb6adca1199a15dd24bfdc003092cba06a/pkg/middleware/auth_proxy/auth_proxy.go#L153-L157

Will prevent team sync from doing what is suppose to do. I was a bit surprised it wasn't documented. 

I did not add any information regarding this behaviour but will happily amend the documentation if this is the case.

**Comment by torkelo** [comment]
Think that is just to stop I happening on every request

**Comment by gotjosh** [comment]
@torkelo agreed - the question from me is whenever this is actually expected from an administrator perspective. 

I'm thinking of:

1. I set up auth proxy team sync
2. I tell my users the next time they open Grafana they'll get placed into the corresponding team
3. Some users do, and some don't (cache dependent) 
4. I'm at confusion as I'm sure my configuration is correct

Moreso, as I wrote the instructions, I faced this same issue. It took me a bit of time to figure out what was going on.

I thought a bit more about this and I'm going to add the footnote as I think users might end up being confused by this behaviour - especially since the key that controls that cache (in case I want to disable it) is `ldap_sync_ttl` which is not straightforward.

**Review by torkelo — APPROVED** [review]

**Pull request merged**